core-data/entities.js

```diff
@@ -23,15 +24,19 @@ export function getEntity( kind, name ) {
 /**
  * Returns the entity's getter method name given its kind and name.
  *
- * @param {string} kind  Entity kind.
- * @param {string} name  Entity name.
+ * @param {string}  kind      Entity kind.
+ * @param {string}  name      Entity name.
+ * @param {string}  prefix    Function prefix.
+ * @param {boolean} usePlural Whether to use the plural form or not.
  *
  * @return {string} Method name
  */
-export const getMethodName = ( kind, name ) => {
+export const getMethodName = ( kind, name, prefix = 'get', usePlural = false ) => {
+	const entity = getEntity( kind, name );
 	const kindPrefix = kind === 'root' ? '' : upperFirst( camelCase( kind ) );
-	const nameSuffix = upperFirst( camelCase( name ) );
-	return `get${ kindPrefix }${ nameSuffix }`;
+	const nameSuffix = upperFirst( camelCase( name ) ) + ( usePlural ? 's' : '' );
+	const suffix = usePlural && entity.plural ? upperFirst( camelCase( entity.plural ) ) : nameSuffix;
```

Comment:
The previous line is wasted work if `usePlural && entity.plural` is truthy. Further, should this be extracted to a utility? Or refactored to logic where we consider a custom `plural` value before doing the `'s'` concatenation.

Aside: On the fence about whether taking on this responsibility from the developer via some utility like [`pluralize`](https://github.com/blakeembrey/pluralize) could be a good thing.

Reply:
I know but I wanted to avoid the embedded ifs or ternaries.